docs/messaging.rst

```diff
@@ -1049,6 +1049,14 @@ multiple cases:
   IPythonQt client) to force a kernel restart to get a clean kernel without
   losing client-side state like history or inlined figures.
 
+A restart should optimally preserve as many resources outside the kernel as
+possible (e.g. only restart the kernel and its subprocesses and not any
+parent processes). That is, ideally a restart should be "in-place". For local
+kernels, there is typically no parent process so a "hard" restart and an in-place
+restart are identical whereas for remote kernels this is not generally the same.
```

Comment:
Great! Maybe worth giving an example, e.g. for a kernel running a container, the container may be kept running and a new kernel process started in the same container without restarting the container. It may also be worth making clear that this is a note for the code _starting_ kernels, since most docs here are about how the kernel should respond to messages.

Reply:
Thoughts on the new wording?